pennylane/interfaces/jax_jit_tuple.py

```diff
@@ -43,71 +44,187 @@ def set_parameters_on_copy_and_unwrap(tapes, params, unwrap=True):
     return tuple(_set_copy_and_unwrap_tape(t, a, unwrap=unwrap) for t, a in zip(tapes, params))
 
 
-def _create_shape_dtype_struct(tape, device):
+def _create_shape_dtype_struct(tape: "qml.tape.QuantumScript", device: "qml.Device"):
     """Auxiliary function for creating the shape and dtype object structure
     given a tape."""
 
-    def process_single_shape(shape, tape_dtype):
-        return jax.ShapeDtypeStruct(tuple(shape), tape_dtype)
-
-    num_measurements = len(tape.measurements)
     shape = tape.shape(device)
-    if num_measurements == 1:
+    if len(tape.measurements) == 1:
         tape_dtype = _numeric_type_to_dtype(tape.numeric_type)
-        return process_single_shape(shape, tape_dtype)
+        return jax.ShapeDtypeStruct(tuple(shape), tape_dtype)
 
     tape_dtype = tuple(_numeric_type_to_dtype(elem) for elem in tape.numeric_type)
-    return tuple(process_single_shape(s, d) for s, d in zip(shape, tape_dtype))
+    return tuple(jax.ShapeDtypeStruct(tuple(s), d) for s, d in zip(shape, tape_dtype))
 
 
-def _tapes_shape_dtype_tuple(tapes, device):
```

Comment:
This function can be reduced to a single line with a tuple + generator.  So it is eliminated in favor of 
```
tuple(_create_shape_dtype_struct(t, device) for t in tapes)
```